### PR TITLE
get_component with net layers, improved

### PR DESCRIPTION
**Pull request by simbilod**
- better docstrings
- complete typings
- no in-place modifications
- separate into smaller functions
- tests

this is a useful function for simulators, and is used in femwell, DEVSIM, and (at least at some point) Elmer/Palace

@joamatab

### CODE REVIEW

**Comment by simbilod** [comment]
@HelgeGehring @nikosavola this would be the new way to define the net layers

File: gplugins/utils/get_component_with_net_layers.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        for name in sublist
+    ]
+    for key in list(new_layerstack.layers.keys()):
+        if key not in layernames_present:
+            del new_layerstack.layers[key]
+
+    return new_layerstack
+
+
+def get_component_with_net_layers(
```
Comment (joamatab) [inline gplugins/utils/get_component_with_net_layers.py:36]: Hi Simon,

the name is not very obvious to me
how can we choose a better name?
How about `get_component_with_new_interface_layers`?

File: gplugins/utils/get_component_with_net_layers.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    component: Component,
+    layerstack: LayerStack,
```
Comment (joamatab) [inline gplugins/utils/get_component_with_net_layers.py:38]: rename to `layer_stack` to be consistent with gdsfactory

File: gplugins/utils/get_component_with_net_layers.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    portnames: list[str],
```
Comment (joamatab) [inline gplugins/utils/get_component_with_net_layers.py:39]: rename to `port_names`

File: gplugins/utils/get_component_with_net_layers.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    delimiter: str = "#",
+    new_layers_init: tuple[int, int] = (10010, 0),
```
Comment (joamatab) [inline gplugins/utils/get_component_with_net_layers.py:41]: rename `new_layers_init` to `new_layers` and define a list of Layers to be used

File: gplugins/utils/get_component_with_net_layers.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
+import copy
+
+import gdsfactory as gf
+from gdsfactory.typings import Component, LayerStack
+
+
+def remove_empty_layerstack_layers(
```
Comment (joamatab) [inline gplugins/utils/get_component_with_net_layers.py:7]: double negation is confusin

How about renaming it to `get_component_layer_stack`? in the end you are returning a layerStack for a particular component

**Review by joamatab — CHANGES_REQUESTED** [review]
thank you Simon

looks great

some minor feedback

File: gplugins/utils/get_component_with_net_layers.py (hunk 1 above)
Comment (simbilod) [reply gplugins/utils/get_component_with_net_layers.py:36]: Sure, although they are not interfaces as of yet. Can we do `get_component_with_new_port_layers`?

File: gplugins/utils/get_component_with_net_layers.py (hunk 3 above)
Comment (simbilod) [reply gplugins/utils/get_component_with_net_layers.py:39]: addressed

File: gplugins/utils/get_component_with_net_layers.py (hunk 1 above)
Comment (joamatab) [reply gplugins/utils/get_component_with_net_layers.py:36]: yes, that sounds good to me

File: gplugins/utils/get_component_with_net_layers.py (hunk 5 above)
Comment (simbilod) [reply gplugins/utils/get_component_with_net_layers.py]: addressed

File: gplugins/utils/get_component_with_net_layers.py (hunk 3 above)
Comment (joamatab) [reply gplugins/utils/get_component_with_net_layers.py:39]: did you push?
i don't see it addressed

Comment (simbilod) [reply gplugins/utils/get_component_with_net_layers.py:39]: Pushed now. Should I squash before we merge?

**Comment by simbilod** [comment]
closed by #140 , thanks you Joaquin

**Pull request closed without merging**